### PR TITLE
Create a Form view to help create a superuser over http

### LINKED ISSUE #162: Build an onboarding flow for new installations
When a new installation starts up for the first time (against an empty database), the user should be directed to an onboarding/setup page. This would ask for:

- Django Super User details:
   - Name
   - Email Address 
   - Password
- `django_site` table domain and name

This should only show up once - after it has been completed it should be locked down and never shown again.

### PR DESCRIPTION
**Pull request by gagantrivedi**
This view can be used to for the initial setup like creating (Only the first)super user, setting the name and domain of the site.

### CODE REVIEW

File: api/users/tests/test_views.py [hunk 1]
```diff
@@ -399,3 +400,51 @@ def test_remove_users_silently_fails_if_user_not_in_group(self):
...
+
+        # When
+        get_response = self.client.get(url)
+        post_response = self.client.get(url)
+
+        # Then
+        assert get_response.status_code == status.HTTP_302_FOUND
+        assert post_response.status_code == status.HTTP_302_FOUND
+
+    def test_returns_403_when_user_exists_and_is_not_logged_in(self):
```
[the remark this answers is not in the document]
Comment (gagantrivedi) [reply api/users/tests/test_views.py]: Yep, it's supposed to say when user is logged in

File: api/users/forms.py [hunk 2]
```diff
@@ -1,6 +1,38 @@
...
+    password = forms.CharField(max_length=32, widget=forms.PasswordInput)
+    site_name = forms.CharField(
+        max_length=50, help_text="A human-readable “verbose” name for the website"
+    )
+    site_domain = forms.CharField(
+        max_length=100,
+        help_text="The fully qualified domain name associated with the website. For example, www.example.com",
+    )
+
+    def create_admin(self):
```
Comment (matthewelwell) [inline api/users/forms.py:27]: I can't see that we've actually tested these form methods?

File: api/users/signals.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import warnings
+
+from django.urls import reverse
+
+
+def warn_insecure(sender, **kwargs):
```
Comment (matthewelwell) [inline api/users/signals.py:6]: Can we write a test for this? I'm not sure if the `caplog` fixture will pick up the `warning` but maybe there's another way to do that if not?

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: api/users/signals.py (hunk 3 above)
Comment (gagantrivedi) [reply api/users/signals.py:6]: The problem is that the warning is triggered by the signal and is emitted before the test, so things like caplog, recwarn will not work with a test, unless we find a way to trigger the signal from within a test.

Comment (gagantrivedi) [reply api/users/signals.py:6]: Or we can just test it like a normal function

Comment (matthewelwell) [reply api/users/signals.py:6]: Yeah, exactly, I just want to test the functionality in the function itself. We can defer the rest to Django :)

File: api/users/forms.py (hunk 2 above)
Comment (gagantrivedi) [reply api/users/forms.py:27]: Added as part of integration tests

File: api/tests/integration/users/test_init_config.py [hunk 4]
```diff
@@ -24,3 +29,62 @@ def test_returns_200_when_no_user_exists(db, django_client):
 
     # Then
     assert response.status_code == status.HTTP_200_OK
+
+
+def test_valid_request_creates_a_superuser(db, django_client):
+    # Given
+    url = reverse("api-v1:users:config-init")
+    user_dict = {
```
Comment (matthewelwell) [inline api/tests/integration/users/test_init_config.py]: I'd say that `user_dict` is a bit misleading here. We can probably just be generic and call this like `data` or `form_data`.

File: api/tests/integration/users/test_init_config.py [hunk 5]
```diff
@@ -24,3 +29,62 @@ def test_returns_200_when_no_user_exists(db, django_client):
...
+
+    # When
+    response = django_client.post(url, data=user_dict)
+
+    # Then
+    assert response.status_code == status.HTTP_200_OK
+    assert FFAdminUser.objects.filter(email=user_dict.get("email")).count() == 1
+
+
+def test_valid_request_updates_site_model(db, django_client):
```
Comment (matthewelwell) [inline api/tests/integration/users/test_init_config.py]: Can we merge this into the test above and just check that both the user and the site model get updated

File: api/tests/integration/users/test_init_config.py [hunk 6]
```diff
@@ -24,3 +29,62 @@ def test_returns_200_when_no_user_exists(db, django_client):
...
+        "password": "test123",
+        "site_name": "test_site",
+        "site_domain": "test.com",
+    }
+
+    # When
+    response = django_client.post(url, data=user_dict)
+
+    # Then
+    assert response.status_code == status.HTTP_200_OK
```
Comment (matthewelwell) [inline api/tests/integration/users/test_init_config.py:70]: We may want a comment explaining why it returns 200?

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: api/tests/integration/users/test_init_config.py (hunk 6 above)
Comment (gagantrivedi) [reply api/tests/integration/users/test_init_config.py:70]: I think it's a pretty standard Django behavior

File: api/users/views.py [hunk 7]
```diff
@@ -14,16 +16,35 @@
...
+
+
+class InitialConfigurationView(PermissionRequiredMixin, FormView):
+    template_name = "users/onboard.html"
+    form_class = InitConfigForm
+    permission_denied_message = (
+        "FAILED TO INIT Configuration. USER(S) ALREADY EXIST IN SYSTEM."
+    )
+
+    def has_permission(self):
```
Comment (matthewelwell) [inline api/users/views.py:37]: Nice! I like this. Keeping all this bespoke functionality nice and neatly in one place 👏

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**